ast/compile.go

```diff
@@ -1740,6 +1740,20 @@ func (c *Compiler) checkKeywordOverrides() {
 	}
 }
 
+func moduleIsRegoV1(mod *Module) bool {
+	switch mod.regoVersion {
+	case RegoUndefined:
+		switch DefaultRegoVersion() {
+		case RegoV1, RegoV0CompatV1:
+			return true
+		default:
+			return false
+		}
+	}
+
+	return mod.regoV1Compatible()
```

Comment:
[nit] the single-case switch could be an if:
```suggestion
	if mod.regoVersion ==  RegoUndefined {
		switch DefaultRegoVersion() {
		case RegoV1, RegoV0CompatV1:
			return true
		}
		return false
	}

	return mod.regoV1Compatible()
```
...but this is code golf and doesn't matter 🙃